### ISSUE TITLE
autohinting may fail unexpectedly when run in mixed mode, suppressing the useful message that a hint might be necessary in the test fixture

### ISSUE THREAD

**Issue by ursjoss** [mockk/mockk#240]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Expected Behavior

If autohinting fails, MockK should indicate that manually setting a hint might be a good idea.

### Current Behavior

If the tests run in mixed mode, type erasure for return types may result in ClassCastExceptions with a message being `null`. This is currently not expected (using `!!` in the method `extractClassName`  in the `JvmAutoHinter`) and results in a `KotlinNullPointerException` instead of simply rethrowing the `ClassCastException` that would later result in the hint about hints :-) being logged.

#### Steps to Reproduce

Unfortunately I have no simple reproducer that reliably illustrates the failing scenario. But looking at the code is already quite convincing given the fact that null messages in ClassCastExceptions may occur in the wild.

#### Context

I have a number of Spek tests using MockK for mocking. All are running successfully when run via IntelliJ spek framework plugin. Two out of those tests keep failing when run through maven, but only on the build server, not reliably on my local development machine. This seems to have to do with running in mixed mode (see e.g.  https://stackoverflow.com/questions/40502576/java-lang-classcastexception-with-null-message-and-cause/40508218).

* MockK version: 1.9
* OS: Windows 10
* Kotlin version: 1.3.20
* JDK version: 1.8.0_192-b12
* Spek Version: 2.0.0-rc1
* Type of test: unit test

#### Stack trace

```
...
kotlin.KotlinNullPointerException
	at io.mockk.impl.recording.JvmAutoHinter.extractClassName(JvmAutoHinter.kt:47)
	at io.mockk.impl.recording.JvmAutoHinter.autoHint(JvmAutoHinter.kt:27)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:36)
	at io.mockk.impl.eval.EveryBlockEvaluator.every(EveryBlockEvaluator.kt:25)
	at io.mockk.MockKDsl.internalEvery(API.kt:92)
	at io.mockk.MockKKt.every(MockK.kt:104)
...
```

I pull request will follow immediatly.

**Comment by oleksiyp** [comment]
Oh really? this NPE is tricky

**Comment by oleksiyp** [comment]
In case of `null` message I output additional information about it:
```
io.mockk.MockKException: Class cast exception happened.
WARN: 'message' property in ClassCastException provided by JVM is null, autohinting is not possible. 
This is most probably happening due to Java optimization enabled. 
You can use `hint` before call or use -XX:-OmitStackTraceInFastThrow to disable this optimization behaviour and make autohinting work. 
For example in gradle use: 

test {
   jvmArgs '-XX:-OmitStackTraceInFastThrow'
}
```

**Comment by ursjoss** [comment]
@oleksiyp Excellent, that makes it very explicit. Thanks!

**Comment by oleksiyp** [comment]
Released [1.9.1](https://github.com/mockk/mockk/releases/tag/1.9.1). Please try it out and close the ticket if everything is fine

**Comment by ursjoss** [comment]
@oleksiyp Thanks, the log are now very clear:

```
[09:12:09]	[Step 5/9] io.mockk.MockKException: 
[09:12:09]	[Step 5/9] Class cast exception happened.
[09:12:09]	[Step 5/9] WARN: 'message' property in ClassCastException provided by JVM is null, autohinting is not possible. 
[09:12:09]	[Step 5/9] This is most probably happening due to Java optimization enabled. 
[09:12:09]	[Step 5/9] You can use `hint` before call or use -XX:-OmitStackTraceInFastThrow to disable this optimization behaviour and make autohiniting work. 
[09:12:09]	[Step 5/9] For example in gradle use: 
[09:12:09]	[Step 5/9] 
[09:12:09]	[Step 5/9] test {
[09:12:09]	[Step 5/9]    jvmArgs '-XX:-OmitStackTraceInFastThrow'
[09:12:09]	[Step 5/9] }
[09:12:09]	[Step 5/9] 	at io.mockk.impl.InternalPlatform.prettifyRecordingException(InternalPlatform.kt:70)
[09:12:09]	[Step 5/9] 	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:62)
[09:12:09]	[Step 5/9] 	at io.mockk.impl.eval.EveryBlockEvaluator.every(EveryBlockEvaluator.kt:30)
[09:12:09]	[Step 5/9] 	at io.mockk.MockKDsl.internalEvery(API.kt:92)
[09:12:09]	[Step 5/9] 	at io.mockk.MockKKt.every(MockK.kt:104)
```

I'm closing the ticket.
